source/LibMultiSense/details/channel.cc

```
@@ -125,15 +125,50 @@ impl::impl(const std::string& address, const RemoteHeadChannel &cameraId) :
     //
     // Create a pool of RX buffers
 
+    uint32_t largeBufferRetry = 0;
     for(uint32_t i=0; i<RX_POOL_LARGE_BUFFER_COUNT; i++)
-        m_rxLargeBufferPool.push_back(new utility::BufferStreamWriter(RX_POOL_LARGE_BUFFER_SIZE));
+    {
+        try {
+            m_rxLargeBufferPool.push_back(new utility::BufferStreamWriter(RX_POOL_LARGE_BUFFER_SIZE));
+            i++;
+        }
+        catch (const std::exception &e) {
+            CRL_DEBUG("Failed to allocate memory (will sleep and try again): %s", e.what());
+            usleep(static_cast<unsigned int> (10000));
+            largeBufferRetry++;
+
+            if (largeBufferRetry >= MAX_BUFFER_ALLOCATION_RETRIES)
+                throw e;
+        }
+    }
+
+    uint32_t smallBufferRetry = 0;
     for(uint32_t i=0; i<RX_POOL_SMALL_BUFFER_COUNT; i++)
-        m_rxSmallBufferPool.push_back(new utility::BufferStreamWriter(RX_POOL_SMALL_BUFFER_SIZE));
+    {
+        try {
+            m_rxSmallBufferPool.push_back(new utility::BufferStreamWriter(RX_POOL_SMALL_BUFFER_SIZE));
+            i++;
+        }
+        catch (const std::exception &e) {
+            CRL_DEBUG("Failed to allocate memory (will sleep and try again): %s", e.what());
+            usleep(static_cast<unsigned int> (10000));
+            smallBufferRetry++;
+
+            if (smallBufferRetry >= MAX_BUFFER_ALLOCATION_RETRIES)
+                throw e;
+        }
+    }
 
     //
     // Bind to the port
 
-    bind();
+    try {
+        bind();
+    } catch (const std::exception& e) {
+        CRL_DEBUG("exception: %s\n", e.what());
+        cleanup();
+        throw e;
```

Do we want throw e here, or do we want to rethrow the exception... there might be some subtle performance or context changes depending on what we do

---

I also figured this was rare enough that we could just throw e